packages/selenium/test/AxeBuilderLegacyTest.cs

```diff
@@ -0,0 +1,401 @@
+
+            var builder = new AxeBuilder(webDriverMock.Object, stubAxeBuilderOptions);
+
+            VerifyExceptionThrown<ArgumentNullException>(() => builder.WithRules(null));
+            VerifyExceptionThrown<ArgumentNullException>(() => builder.DisableRules(null));
+            VerifyExceptionThrown<ArgumentNullException>(() => builder.WithTags(null));
+            VerifyExceptionThrown<ArgumentNullException>(() => builder.Include(null));
+            VerifyExceptionThrown<ArgumentNullException>(() => builder.Exclude(null));
+            VerifyExceptionThrown<ArgumentNullException>(() => builder.WithOptions(null));
+        }
+
+        [Test]
+        public void LegacyShouldThrowIfEmptyParameterPassed()
+        {
+            var values = new string[] { "val1", "" };
+
+            SetupVerifiableAxeInjectionCall();
+
+            var builder = new AxeBuilder(webDriverMock.Object, stubAxeBuilderOptions);
+
+            VerifyExceptionThrown<ArgumentException>(() => builder.WithRules(values));
+            VerifyExceptionThrown<ArgumentException>(() => builder.DisableRules(values));
+            VerifyExceptionThrown<ArgumentException>(() => builder.WithTags(values));
+            VerifyExceptionThrown<ArgumentException>(() => builder.Include(values));
+            VerifyExceptionThrown<ArgumentException>(() => builder.Exclude(values));
+        }
+
+        [Test]
+        public void LegacyShouldThrowIfDeprecatedOptionsIsUsedWithNewOptionsApis()
+        {
+            SetupVerifiableAxeInjectionCall();
+
+            var builder = new AxeBuilder(webDriverMock.Object, stubAxeBuilderOptions);
+#pragma warning disable CS0618
+            builder.Options = "{xpath:true}";
+#pragma warning restore CS0618
+
+            VerifyExceptionThrown<InvalidOperationException>(() => builder.WithRules("rule-1"));
+            VerifyExceptionThrown<InvalidOperationException>(() => builder.DisableRules("rule-1"));
+            VerifyExceptionThrown<InvalidOperationException>(() => builder.WithTags("tag1"));
+            VerifyExceptionThrown<InvalidOperationException>(() => builder.WithOptions(new AxeRunOptions() { Iframes = true }));
+        }
+
+        private void VerifyExceptionThrown<T>(Action action) where T : Exception
+        {
+            action.Should().Throw<T>();
+        }
+
+        private void VerifyAxeResult(AxeResult result)
+        {
+            result.Should().NotBeNull();
+            result.Inapplicable.Should().NotBeNull();
+            result.Incomplete.Should().NotBeNull();
+            result.Passes.Should().NotBeNull();
+            result.Violations.Should().NotBeNull();
+
+            result.Inapplicable.Length.Should().Be(0);
+            result.Incomplete.Length.Should().Be(0);
+            result.Passes.Length.Should().Be(0);
+            result.Violations.Length.Should().Be(0);
+        }
+
+        private static void SetupVerifiableAxeInjectionCall()
```

Comment:
Most of these tests are failing because this is missing a few bits of required mock setup you added to `AxeBuilderTest.cs` , particularly:

```csharp
            webDriverMock
                .Setup(d => d.FindElements(It.IsAny<By>()))
                .Returns(new List<IWebElement>().AsReadOnly());
            webDriverMock
                .Setup(d => d.SwitchTo())
                .Returns(targetLocatorMock.Object);
```